ESQL: TS Command, grouping on value that is sometimes null causes unexpected partial results

In this query:
```
ts k8* 
| STATS count(rate(network.total_cost)) 
BY region, bucket(@timestamp,1hour)
```
`region:keyword` is not present in the `k8s-downsampled` index, which I suspect is the source of this error.


Error:
```
    java.lang.AssertionError: unexpected partial results: _clusters={details={(local)={_shards={total=2, failed=2, successful=0, skipped=0}, took=267, indices=k8*, failures=[{node=null, reason={reason=class org.elasticsearch.compute.data.DocBlock cannot be cast to class org.elasticsearch.compute.data.BytesRefBlock (org.elasticsearch.compute.data.DocBlock and org.elasticsearch.compute.data.BytesRefBlock are in unnamed module of loader java.net.URLClassLoader @2213639b), type=class_cast_exception}, index=k8s, shard=0}], status=partial}}}
    Expected: (null or is <false>)
         but: was <true>
```

--- a/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/DimensionValuesByteRefGroupingAggregatorFunction.java
+++ b/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/DimensionValuesByteRefGroupingAggregatorFunction.java
@@ -73,7 +73,7 @@ public final class DimensionValuesByteRefGroupingAggregatorFunction implements G
 
     @Override
     public AddInput prepareProcessRawInputPage(SeenGroupIds seenGroupIds, Page page) {
-        BytesRefBlock valuesBlock = page.getBlock(0);
+        BytesRefBlock valuesBlock = page.getBlock(channel);
         if (valuesBlock.areAllValuesNull()) {
             return new AddInput() {
                 @Override

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/aggregate/DimensionValuesByteRefGroupingAggregatorFunctionTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/aggregate/DimensionValuesByteRefGroupingAggregatorFunctionTests.java
@@ -11,6 +11,7 @@ import org.apache.lucene.util.BytesRef;
 import org.elasticsearch.compute.aggregation.AggregatorMode;
 import org.elasticsearch.compute.aggregation.DimensionValuesByteRefGroupingAggregatorFunction;
 import org.elasticsearch.compute.aggregation.blockhash.BlockHash;
+import org.elasticsearch.compute.data.Block;
 import org.elasticsearch.compute.data.BlockFactory;
 import org.elasticsearch.compute.data.BytesRefBlock;
 import org.elasticsearch.compute.data.ElementType;
@@ -23,6 +24,7 @@ import org.elasticsearch.compute.operator.PageConsumerOperator;
 import org.elasticsearch.compute.test.CannedSourceOperator;
 import org.elasticsearch.compute.test.ComputeTestCase;
 import org.elasticsearch.compute.test.OperatorTestCase;
+import org.elasticsearch.compute.test.RandomBlock;
 import org.elasticsearch.compute.test.TestDriverFactory;
 import org.hamcrest.Matchers;
 
@@ -43,6 +45,8 @@ public class DimensionValuesByteRefGroupingAggregatorFunctionTests extends Compu
         int numPages = between(1, 10);
         List<Page> pages = new ArrayList<>(numPages);
         Map<Integer, List<BytesRef>> expectedValues = new HashMap<>();
+        int prefixBlocks = between(0, 2);
+        int suffixBlocks = between(0, 2);
         for (int i = 0; i < numPages; i++) {
             int positions = between(1, 100);
             try (
@@ -70,20 +74,35 @@ public class DimensionValuesByteRefGroupingAggregatorFunctionTests extends Compu
                     groups.appendInt(group);
                     expectedValues.putIfAbsent(group, values);
                 }
-                pages.add(new Page(valuesBuilder.build(), groups.build().asBlock()));
+                BytesRefBlock valuesBlock = valuesBuilder.build();
+                IntBlock groupsBlock = groups.build().asBlock();
+                Block[] blocks = new Block[prefixBlocks + 2 + suffixBlocks];
+                for (int b = 0; b < prefixBlocks; b++) {
+                    blocks[b] = RandomBlock.randomBlock(RandomBlock.randomElementType(), positions, false, 1, 1, 0, 0).block();
+                }
+                blocks[prefixBlocks] = valuesBlock;
+                blocks[prefixBlocks + 1] = groupsBlock;
+                for (int b = 0; b < suffixBlocks; b++) {
+                    blocks[prefixBlocks + 2 + b] = RandomBlock.randomBlock(RandomBlock.randomElementType(), positions, false, 1, 1, 0, 0)
+                        .block();
+                }
+                pages.add(new Page(blocks));
             }
         }
         var aggregatorFactory = new DimensionValuesByteRefGroupingAggregatorFunction.FunctionSupplier().groupingAggregatorFactory(
-            randomFrom(AggregatorMode.INITIAL, AggregatorMode.SINGLE),
-            List.of(0)
+            randomFrom(AggregatorMode.INITIAL, AggregatorMode.SINGLE, AggregatorMode.FINAL),
+            List.of(prefixBlocks)
         );
         final List<BlockHash.GroupSpec> groupSpecs;
         if (randomBoolean()) {
             // Use IntBlockHash; reserves 0 for null values
-            groupSpecs = List.of(new BlockHash.GroupSpec(1, ElementType.INT));
+            groupSpecs = List.of(new BlockHash.GroupSpec(prefixBlocks + 1, ElementType.INT));
         } else {
             // Use PackedValuesBlockHash; does not reserve 0 for null values
-            groupSpecs = List.of(new BlockHash.GroupSpec(1, ElementType.INT), new BlockHash.GroupSpec(1, ElementType.INT));
+            groupSpecs = List.of(
+                new BlockHash.GroupSpec(prefixBlocks + 1, ElementType.INT),
+                new BlockHash.GroupSpec(prefixBlocks + 1, ElementType.INT)
+            );
         }
         HashAggregationOperator hashAggregationOperator = new HashAggregationOperator(
             List.of(aggregatorFactory),

